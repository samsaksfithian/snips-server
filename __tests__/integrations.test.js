/* eslint-disable no-console */
require('dotenv').config();
const request = require('supertest');
const app = require('../app');
const db = require('../db');
const dbInit = require('../db/init');

beforeAll(async () => {
  await dbInit.init(true);
});

describe('Snippets', () => {
  describe('GET /api/snippets', () => {
    it('should get all of the snippets', async () => {
      // test the /api/snippets route
      const response = await request(app).get('/api/snippets');
      expect(response.body.length).toBe(2);
      expect(response.status).toBe(200);
      expect(response.error).toBeFalsy();
      expect(response.body).toMatchSnapshot();
    });
  });
});

afterAll(() => {
  // close the db pool
  db.end();
});
