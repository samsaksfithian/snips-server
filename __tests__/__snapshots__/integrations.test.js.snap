// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippets GET /api/snippets should get all of the snippets 1`] = `
Array [
  Object {
    "author": "Dean",
    "code": "4 + 4",
    "description": "This is how you add",
    "favorites": 0,
    "id": 1,
    "language": "Algebra",
    "title": "Addition",
  },
  Object {
    "author": "Emily",
    "code": "console.log(\\"Hello world!\\");",
    "description": "My first program",
    "favorites": 0,
    "id": 2,
    "language": "javascript",
    "title": "Hello World",
  },
]
`;
